### PR TITLE
poll to wait for a pod before starting port forward

### LINKED ISSUE #2883: gitops run: race condition in setting up port forwarding
At first run, it's possible for the flux resources to be healthy and available, but for the pods not to be.

To reproduce, grab a cluster without podinfo running. Create a new directory with a file with this in it:
```
---
apiVersion: v1
kind: Namespace
metadata:
  name: dev
---
apiVersion: source.toolkit.fluxcd.io/v1beta2
kind: HelmRepository
metadata:
  name: podinfo
  namespace: dev
spec:
  interval: 1m0s
  url: https://stefanprodan.github.io/podinfo
---
apiVersion: helm.toolkit.fluxcd.io/v2beta1
kind: HelmRelease
metadata:
  name: podinfo
  namespace: dev
spec:
  chart:
    spec:
      chart: podinfo
      reconcileStrategy: ChartVersion
      sourceRef:
        kind: HelmRepository
        name: podinfo
  interval: 1m0s
  targetNamespace: dev
```
and run gitops run as

`gitops beta run ./podinfo --no-session --port-forward namespace=dev,resource=svc/dev-podinfo,port=9898:9898`

The first run you should see `✗ Error getting pod from specMap: no running pods found for service`. Re-save the file, and it'll update and the port forwarding will work.

### PR DESCRIPTION
**Pull request by chanwit**
Fixes #2883

Signed-off-by: Chanwit Kaewkasi <chanwit@weave.works>

### CODE REVIEW

File: cmd/gitops/beta/run/cmd.go [hunk 1]
```diff
@@ -897,12 +899,26 @@ func runCommandWithoutSession(cmd *cobra.Command, args []string) error {
...
+
+							log.Waitingf("Waiting for a pod from specMap: %v", podErr)
+							return false, nil
+						}); pollErr != nil {
+							log.Failuref("Waiting for a pod from specMap: %v", pollErr)
 						}
 
-						if pod != nil {
+						if pod == nil {
+							log.Failuref("Error getting pod from specMap")
```
Comment (makkes) [inline cmd/gitops/beta/run/cmd.go:920]: shouldn't we return here?

File: cmd/gitops/beta/run/cmd.go [hunk 2]
```diff
@@ -897,12 +899,26 @@ func runCommandWithoutSession(cmd *cobra.Command, args []string) error {
...
+						if pollErr := wait.PollImmediate(2*time.Second, flags.Timeout, func() (bool, error) {
+							pod, podErr = run.GetPodFromResourceDescription(thisCtx, namespacedName, specMap.Kind, kubeClient)
+							if pod != nil && podErr == nil {
+								return true, nil
+							}
+
+							log.Waitingf("Waiting for a pod from specMap: %v", podErr)
+							return false, nil
+						}); pollErr != nil {
+							log.Failuref("Waiting for a pod from specMap: %v", pollErr)
```
Comment (makkes) [inline cmd/gitops/beta/run/cmd.go:916]: shouldn't we return here?

**Review by makkes — COMMENTED** [review]
lgtm, just 2 questions.

File: cmd/gitops/beta/run/cmd.go (hunk 2 above)
Comment (chanwit) [reply cmd/gitops/beta/run/cmd.go:916]: we cannot.

returning means stopping the go routine.
we have to log and wait for the next trigger.

**Review by opudrovs — APPROVED** [review]
LGTM! 🎉

**Review by makkes — APPROVED** [review]

**Pull request merged**